lib/molinillo/resolution.rb

```diff
@@ -382,21 +387,17 @@ def unwind_details_for_conflict(conflict, existing_unwind_details = nil)
           until grandparent_r.nil?
             partial_tree.unshift(grandparent_r)
             requirement_state = find_state_for(grandparent_r)
-            candidate_index = states.index(requirement_state)
-            break unless candidate_index
-            candidate_details =
-              UnwindDetails.new(candidate_index, grandparent_r, partial_tree, binding_requirements, trees)
-            break unless candidate_details > unwind_details
             if requirement_state.possibilities.any? { |set| !set.dependencies.include?(parent_r) }
-              unwind_details = candidate_details
-              break
+              candidate_index = states.index(requirement_state)
+              unwind_details <<
+                UnwindDetails.new(candidate_index, grandparent_r, partial_tree, binding_requirements, trees)
             end
             parent_r = grandparent_r
             grandparent_r = parent_of(parent_r)
           end
         end
 
-        unwind_details
+        unwind_details.sort
```

Comment:
is it possible this array won't be used, and we could maybe `sort!` it when first accessed?